registry/proxy/scheduler/scheduler.go

```diff
@@ -0,0 +1,260 @@
+package scheduler
+
+// todo: move to own package and make constructor
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/filesystem"
+)
+
+// onTTLExpiryFunc is called when a repositories' TTL expires
+type expiryFunc func(string) error
+
+const (
+	entryTypeBlob = iota
+	entryTypeManifest
+)
+
+// schedulerEntry represents an entry in the scheduler
+// fields are exported for serialization
+type schedulerEntry struct {
+	Key        string
+	ExpiryDate time.Time
+	EntryType  int
+}
+
+var theScheduler *ttlExpirationScheduler
+
+func init() {
+	theScheduler = new(context.Background(), filesystem.New("/tmp"), "/ttl.json")
+}
+
+// ttlExpirationScheduler is a scheduler used to perform actions
+// when TTLs expire
+type ttlExpirationScheduler struct {
+	entries  map[string]schedulerEntry
+	addChan  chan schedulerEntry
+	stopChan chan bool
+
+	driver          driver.StorageDriver
```

Comment:
Why use the storage driver here if we are just doing local filesystem operations?